src/spatial_ref/srs.rs

```diff
@@ -303,6 +314,111 @@ impl SpatialRef {
         }
     }
 
+    #[cfg(major_ge_3)]
+    pub fn get_name(&self) -> Result<String> {
+        let c_ptr = unsafe { gdal_sys::OSRGetName(self.0) };
+        if c_ptr.is_null() {
+            return Err(_last_null_pointer_err("OSRGetName"));
+        }
+        Ok(_string(c_ptr))
+    }
+
+    pub fn get_angular_units_name(&self) -> Result<String> {
+        let mut c_ptr = ptr::null_mut();
+        unsafe { gdal_sys::OSRGetAngularUnits(self.0, &mut c_ptr) };
+        if c_ptr.is_null() {
+            return Err(_last_null_pointer_err("OSRGetAngularUnits"));
+        }
+        Ok(_string(c_ptr))
+    }
+
+    pub fn get_angular_units(&self) -> f64 {
```

Comment:
So if I want to get both the name and value I need to make two calls like this right?

```
let name = srs.get_angular_units_name();
let value = srs.get_angular_units();
```

Alternatively, you could consider making the api something like:
```
let (name, value) = srs.get_angular_units()
let name = srs.get_angular_units_name();
let value = srs.get_angular_units_value();
```

(I don't have any authority in whether or not this gets merged, I'm just a gdal crate user that watches PR's come by)

Reply:
uh well i guess both ways are fine. I would suggest to keep it this way to avoid mixing Results.

Reply:
It tooks me some time to end up with this solution but  many of the use case I have faced up involved returning either only the name or only the result for calculation, since returning the name implies allocation it considered that it was better to have a zero cost way to get the units factor value (note: this is also the choice of the python bindings).